qemu/tests/qmp_command.py

```diff
@@ -200,41 +162,42 @@ def check_result(qmp_o, output=None, exception_list=""):
         qmp_port = qmp_ports[0]
     else:
         test.error("Incorrect configuration, no QMP monitor found.")
-    hmp_ports = vm.get_monitors_by_type('human')
-    if hmp_ports:
-        hmp_port = hmp_ports[0]
-    else:
-        test.error("Incorrect configuration, no QMP monitor found.")
     callback = {"host_cmd": lambda cmd: process.system_output(cmd, shell=True).decode(),
                 "guest_cmd": session.cmd_output,
-                "monitor_cmd": hmp_port.send_args_cmd,
                 "qmp_cmd": qmp_port.send_args_cmd}
 
     def send_cmd(cmd):
-        """ Helper to execute command on ssh/host/monitor """
+        """ Helper to execute command on host/ssh guest/qmp monitor """
         if cmd_type in callback.keys():
             return callback[cmd_type](cmd)
         else:
             test.error("cmd_type is not supported")
 
     pre_cmd = params.get("pre_cmd")
     qmp_cmd = params.get("qmp_cmd")
-    cmd_type = params.get("event_cmd_type")
     post_cmd = params.get("post_cmd")
+    cmd_type = params.get("event_cmd_type")
     result_check = params.get("cmd_result_check")
-    cmd_return_value = params.get("cmd_return_value")
+    strict_match = params.get("strict_match", "yes") == 'yes'
+    expect_o = eval(params.get("cmd_return_value", "[]"))
     exception_list = params.get("exception_list", "")
 
-    # Pre command
+    # Pre qmp command
     if pre_cmd is not None:
         logging.info("Run prepare command '%s'.", pre_cmd)
         pre_o = send_cmd(pre_cmd)
+        if not isinstance(pre_o, list):
+            pre_o = [pre_o]
```

Comment:
Why need pre_o to be list?

Reply:
It is not must, just keep consistent with `qmp_o` and  `post_o` before.  Delete this redundant operation, thanks!